Orange/data/io.py

```diff
@@ -1,384 +1,537 @@
-        domain_vars = data.domain.variables + data.domain.metas
-        # first line
-        f.write("\t".join([str(j.name) for j in domain_vars]))
-        f.write("\n")
-
-        # second line
-        # TODO Basket column.
-        t = {"ContinuousVariable": "c", "DiscreteVariable": "d",
-             "StringVariable": "string", "Basket": "basket"}
-
-        f.write("\t".join([t[type(j).__name__] for j in domain_vars]))
-        f.write("\n")
-
-        # third line
-        m = list(data.domain.metas)
-        c = list(data.domain.class_vars)
-        r = []
-        for i in domain_vars:
-            r1 = ["{}={}".format(k, v).replace(" ", "\\ ")
-                  for k, v in i.attributes.items()]
-            if i in m:
-                r1.append("m")
-            elif i in c:
-                r1.append("class")
-            r.append(" ".join(r1))
-        f.write("\t".join(r))
-        f.write("\n")
-
-        # data
-        # noinspection PyBroadException
-        try:
-            cls._write_fast(f, data)
-        except:
-            domain_vars = [data.domain.index(var) for var in domain_vars]
-            for i in data:
-                f.write("\t".join(str(i[j]) for j in domain_vars) + "\n")
-        f.close()
+            if 1 == len(headers):
+                HEADER1_FLAG_SEP = '#'
+                # First row format either:
+                #   1) delimited column names
+                #   2) -||- with type and flags prepended, separated by #,
+                #      e.g. d#sex,c#age,cC#IQ
+                _flags, names = zip(*[i.split(HEADER1_FLAG_SEP, 1) if HEADER1_FLAG_SEP in i else ('', i)
+                                      for i in headers[0]])
+                names = list(names)
+            elif 2 == len(headers):
+                names, _flags = map(list, headers)
+            else:
+                # Use heuristics for everything
+                names, _flags = [], []
+            types = [''.join(filter(str.isupper, flag)).lower() for flag in _flags]
+            flags = [Flags.join(filter(str.islower, flag)) for flag in _flags]
 
-    def write(self, filename, data):
-        self.write_file(filename, data)
+        # Determine maximum row length
+        rowlen = max(map(len, (names, types, flags)))
 
+        def _equal_length(lst):
+            lst.extend(['']*(rowlen - len(lst)))
+            return lst
 
-@FileFormats.register("Comma-separated file", ".csv")
-class TxtFormat:
-    MISSING_VALUES = frozenset({"", "NA", "?"})
+        # Ensure all data is of equal width in a column-contiguous array
+        data = np.array([_equal_length(list(row)) for row in data if any(row)],
+                        copy=False, dtype=object, order='F')
+
+        # Data may actually be longer than headers were
+        try: rowlen = data.shape[1]
+        except IndexError: pass
+        else:
+            for lst in (names, types, flags):
+                _equal_length(lst)
+
+        NAMEGEN = namegen('_')
```

Comment:
Won't these name-generated variables (`_0`, `_1`, `_2`, ...) clash with one another if multiple sources without headers are loaded and these variables try to be unified with the existing in the `coltype.make()` call below? Wouldn't a global `NAMEGEN` make more sense?